document/src/main/java/com/yahoo/document/DocumentUpdate.java

```diff
@@ -55,7 +58,7 @@
      * @param docType the document type that this update is valid for
      */
     public DocumentUpdate(DocumentType docType, DocumentId docId) {
-        this(docType, docId, new ArrayList<FieldUpdate>());
+        this(docType, docId, new HashMap<Integer, FieldUpdate>());
```

Comment:
With regards to the comments concerning ordering of elements to support the deprecated API, would using `LinkedHashMap` be a safer option?

Reply:
Not an issue any more since there is a backing list